mender.go

```diff
@@ -15,28 +15,30 @@ package main
 
 import (
 	"bufio"
-	"encoding/json"
-	"errors"
-	"io/ioutil"
 	"os"
 	"strings"
 	"time"
 
 	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
 )
 
 type Controler interface {
-	Bootstrap() error
-	TransitionState() MenderState
+	Bootstrap() menderError
 	GetCurrentImageID() string
-	GetDaemonConfig() daemonConfig
-	GetUpdaterConfig() httpsClientConfig
-	LastError() error
+	GetUpdatePollInterval() time.Duration
+	HasUpgrade() (bool, menderError)
+	CheckUpdate() (*UpdateResponse, menderError)
```

Comment:
I disagree. It's an interface of the controller that can then be consumed by someone else (states in this case). If I move that to a client of some sort, then the code that drives the controller will have to interact with that client.